### DISCUSSION TITLE
[HW1] Average pooling for padded sequences

### DISCUSSION THREAD

**Discussion by silvershine157** [seominjoon/kaist-ai605#4] [Q&A] [answered]
Hello, I have a question about P3.1 of HW1.
It seems that we have to train our Baseline model with batch size bigger than 1,
Since forward() method of Baseline takes a tensor as input, I assume we have to pad the sentences to match the lengths of sentences in a minibatch.
However, it seems that mean pooling operation (avg = out.mean(1)) does not consider how the input is padded.
Is it okay to modify the Baseline class to resolve this issue?
Thank you.

**Comment by seominjoon** [comment, marked as answer]
Yes, of course. Please modify it freely. The objective is to create a functional model and other things are secondary.

 There are several methods for collating sequences of different lengths and padding + masking is probably the dominant method, so I definitely encourage you to use it. But it is not the only way, so please feel free to explore.

**Reply by silvershine157** [reply]
I see, thank you professor!
